### PR TITLE
Fireworks Fix

### PR DESCRIPTION
**Pull request by TheAbsolutionism**
### Description
This PR aims to fix the firework event when using a color type
Also adds a colors event-value

---
**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:** #6091

### CODE REVIEW

File: src/test/java/org/skriptlang/skript/test/tests/syntaxes/events/EvtFireworkTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import org.bukkit.event.Event;
+import org.bukkit.event.entity.FireworkExplodeEvent;
+import org.bukkit.inventory.meta.FireworkMeta;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
```
Comment (Efnilite) [inline src/test/java/org/skriptlang/skript/test/tests/syntaxes/events/EvtFireworkTest.java:21]:
```suggestion
```

File: src/test/java/org/skriptlang/skript/test/tests/syntaxes/events/EvtFireworkTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+public class EvtFireworkTest extends SkriptJUnitTest {
+
+	private EntityType entityType;
```
Comment (Efnilite) [inline src/test/java/org/skriptlang/skript/test/tests/syntaxes/events/EvtFireworkTest.java:24]:
```suggestion
	private EntityType entityType;

```

File: src/test/java/org/skriptlang/skript/test/tests/syntaxes/events/EvtFireworkTest.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		} else {
+			entityType = EntityType.valueOf("FIREWORK");
+		}
+	}
+
+	@Test
+	public void callEvents() {
+		List<Event> events = new ArrayList<>();
+		for (SkriptColor color : SkriptColor.values()) {
+			Firework firework = (Firework) getTestWorld().spawnEntity(getTestLocation(), entityType);
```
Comment (Efnilite) [inline src/test/java/org/skriptlang/skript/test/tests/syntaxes/events/EvtFireworkTest.java:38]: shouldn't these be removed in `@After`?

File: src/main/java/ch/njol/skript/util/SkriptColor.java [hunk 4]
```diff
@@ -178,6 +178,8 @@ public static SkriptColor fromBukkitColor(org.bukkit.Color color) {
 		for (SkriptColor c : colors) {
 			if (c.asBukkitColor().equals(color))
 				return c;
+			else if (c.asDyeColor().getFireworkColor().equals(color))
+				return c;
```
Comment (Efnilite) [inline src/main/java/ch/njol/skript/util/SkriptColor.java:182]:
```suggestion
			if (c.asBukkitColor().equals(color) || c.asDyeColor().getFireworkColor().equals(color))
				return c;
```

File: src/main/java/ch/njol/skript/events/EvtFirework.java [hunk 5]
```diff
@@ -41,15 +22,15 @@ public class EvtFirework extends SkriptEvent {
...
-					.examples("on firework explode",
-							"on firework exploding colored red, light green and black",
+					.examples("on firework explode:",
+								"\tif event-colors contains red:",
+							"on firework exploding colored red, light green and black:",
 							"on firework explosion colored light green:",
-							"	broadcast \"A firework colored %colors% was exploded at %location%!\"")//TODO fix 
-					.since("2.4");
+								"\tbroadcast \"A firework colored %colors% was exploded at %location%!\"")
+					.since("2.4, INSERT VERSION (fix colored)");
```
Comment (Efnilite) [inline src/main/java/ch/njol/skript/events/EvtFirework.java:30]:
```suggestion
					.since("2.4");
```

**Review by Efnilite — CHANGES_REQUESTED** [review]

File: src/test/java/org/skriptlang/skript/test/tests/syntaxes/events/EvtFireworkTest.java (hunk 3 above)
Comment (TheAbsolutionism) [reply src/test/java/org/skriptlang/skript/test/tests/syntaxes/events/EvtFireworkTest.java:38]: Doesnt the `FireworkExplodeEvent` cause them to be removed? I could be wrong, I can add it still to ensure they do though

File: src/main/java/ch/njol/skript/expressions/ExprFireworkEffect.java [hunk 6]
```diff
@@ -87,10 +87,10 @@ protected FireworkEffect[] get(Event e) {
 		FireworkEffect.Builder builder = FireworkEffect.builder().with(type);
 		
 		for (Color colour : color.getArray(e))
-			builder.withColor(colour.asBukkitColor());
+			builder.withColor(colour.asDyeColor().getFireworkColor());
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/expressions/ExprFireworkEffect.java:90]: Transforming the color into a dye color removes the RGB support Fireworks and SkriptColor can support.

This should be described in the description of the syntax that the color comparison will only work on vanilla Minecraft dye colors, or support should be added for RGB color comparison.

**Review by TheLimeGlass — CHANGES_REQUESTED** [review]

**Review by Efnilite — APPROVED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprFireworkEffect.java (hunk 6 above)
Comment (TheAbsolutionism) [reply src/main/java/ch/njol/skript/expressions/ExprFireworkEffect.java:90]: Hey, so I'm just curious what you think would be best, 
adding a note in the description explaining the `%color%` will only work when comparing to a vanilla mc dye colors
or somehow adding support for RGB in the syntax?

**Review by sovdeeth — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/events/EvtFirework.java [hunk 7]
```diff
@@ -58,16 +40,22 @@ public boolean init(Literal<?>[] args, int matchedPattern, ParseResult parseResu
...
-				.map(color -> color.asBukkitColor())
-				.collect(Collectors.toList());
-		FireworkMeta meta = ((FireworkExplodeEvent)e).getEntity().getFireworkMeta();
+		List<org.bukkit.Color> colours = Arrays.stream(colors.getArray(event))
+				.map(color -> {
+					if (color instanceof ColorRGB)
+						return color.asBukkitColor();
+					return color.asDyeColor().getFireworkColor();
+				})
+				.toList();
```
Comment (Fusezion) [inline src/main/java/ch/njol/skript/events/EvtFirework.java:57]:
```suggestion
		List<org.bukkit.Color> colours = colors.stream(event)
				.map(color -> {
					if (color instanceof ColorRGB)
						return color.asBukkitColor();
					return color.asDyeColor().getFireworkColor();
				})
				.toList();
```

**Review by Fusezion — CHANGES_REQUESTED** [review]
Just small change to remove some needless conversions

**Review by Fusezion — APPROVED** [review]
image

**Review by TheLimeGlass — APPROVED** [review]

**Review by Moderocky — APPROVED** [review]
It seems ok.

**Pull request merged**